lib/backend/test/suite.go

```diff
@@ -561,17 +566,41 @@ func testEvents(t *testing.T, newBackend Constructor) {
 	e = requireEvent(t, watcher, types.OpPut, item.Key, eventTimeout)
 	require.Equal(t, item.Value, e.Item.Value)
 
-	// Wait a few seconds for the item to expire.
-	clock.Advance(3 * time.Second)
-
 	// Make sure item has been removed.
-	_, err = uut.Get(ctx, item.Key)
-	require.Error(t, err)
+	require.Eventually(t, func() bool {
+		_, err = uut.Get(ctx, item.Key)
+		return trace.IsNotFound(err)
+	}, time.Second*4, time.Millisecond*200, "Failed to ensure that item %q has been deleted", item.Key)
```

Comment:
curious why `require.Eventually` to replace `clock.Advance`?

Reply:
The 3 tests in DynamoDB test suite were failing. The `clock.Advance` doesn't affect the DynamoDB clock so enforcing items to expire by calling `clock.Advance` doesn't work.  So in this case we need to relay on real clock.

Reply:
cool. 

for the `Get` failure at least, seems the `record` is our own type. Maybe we can pass the clock to `isExpired` function or assign the clock to the object
https://github.com/gravitational/teleport/blob/cda00f3e798db6582ee2885da96900ed5490a381/lib/backend/dynamo/dynamodbbk.go#L866-L871

I am ok with `Eventually` too. was just curious.

Reply:
I have found 3 tests failing from DynamoDB suite.  the DynamoDB item uses time.Now clock to check if item was expired thus calling `clock.Advance` doesn't enforce item to expired. 

Though after a second thought i have aligned time DynamoDB item logic to leverage the `b.Clock`